### PR TITLE
 [LIVY-622][LIVY-623][LIVY-624][LIVY-625][THRIFT][FOLLOWUP] Use ResultSet in catalog operations

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

This is a followup of #194 which addresses all the remaining concerns. The main changes are:

 - reverting the introduction of a state specific for catalog operations;
 - usage of `ResultSet` to send over the wire the data for catalog operations too.

## How was this patch tested?

existing modified UTs

### CODE REVIEW

**Comment by mgaido91** [comment]
cc @jerryshao @yantzu

**Comment by mgaido91** [comment]
If there are no comments I'm going to merge in a couple of days.

**Comment by jerryshao** [comment]
CC @yiheng , please take a look.

File: thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java [hunk 1]
```diff
@@ -59,6 +59,52 @@
...
+          sparkDt = DataTypes.BooleanType;
+          break;
+        case SHORT:
+          sparkDt = DataTypes.IntegerType;
+          break;
+        case INTEGER:
+          sparkDt = DataTypes.IntegerType;
+          break;
+        case LONG:
+          sparkDt = DataTypes.FloatType;
```
Comment (yiheng) [inline thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java]: Should it be `DataTypes.LongType` ?

File: thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java [hunk 2]
```diff
@@ -59,6 +59,52 @@
...
+    StructField[] fields = new StructField[types.length];
+    int idx = 0;
+    for (DataType dt : types) {
+      org.apache.spark.sql.types.DataType sparkDt = null;
+      switch (dt) {
+        case BOOLEAN:
+          sparkDt = DataTypes.BooleanType;
+          break;
+        case BYTE:
+          sparkDt = DataTypes.BooleanType;
```
Comment (yiheng) [inline thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java]: Should it be `DataTypes.ByteType`?

File: thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java [hunk 3]
```diff
@@ -59,6 +59,52 @@
...
+          break;
+        case SHORT:
+          sparkDt = DataTypes.IntegerType;
```
Comment (yiheng) [inline thriftserver/session/src/main/java/org/apache/livy/thriftserver/session/SparkUtils.java]: Should it be `DataTypes.ShortType`?

**Comment by yiheng** [comment]
+1

**Comment by jerryshao** [comment]
Looks like there's some conflicts need to be fixed.

**Comment by mgaido91** [comment]
Thank you all. Merging to master.

**Pull request closed without merging**